backend/compact-connect/lambdas/provider-data-v1/tests/unit/test_data_model/test_paginated.py

```diff
@@ -110,22 +116,28 @@ def get_something(*args, **kwargs):
             kwarg1='baf'
         )
 
-        self.assertEqual(
-            {
-                # 12 items will have been returned from queries internally, but only 10 make it out
-                # to fill out the pageSize
-                'items': [ProviderRecordSchema().load(self._item)] * 10,
-                'pagination': {
-                    'pageSize': 10,
-                    'lastKey': b64encode(
-                        json.dumps({'pk': self._item['pk'], 'sk': self._item['sk']}).encode('utf-8')
-                    ).decode('ascii'),
-                    'prevLastKey': last_key
-                },
-            },
-            resp
-        )
-        # 2 calls, each returning 6 items, will fill out the page size.
+        # We are requesting 10 users, starting with exclusive key 1, and filtering out all odds client-side. This
+        # should result in three queries to the DB, with the last record included in the response having a pk of 20:
+        #
+        # | Query | DB sequence | PK | Ret Sequence |  Filter   | last_key |
```

Comment:
I'm confused by this table. What is DB sequence and return Sequence? 

Also, where are the records being filtered in the tests? I can't seem to find that.

Reply:
Oh, I didn't end up doing client-side filtering here so I should remove that part of the comment.

As for the table, this is trying to walk through the sequence of queries to the DB vs the items our backend returns to the client, to demonstrate expected internal behavior. `Query` is the number of the query the backend makes to the DB. `DB sequence` is the position in the returned records each item has for that query:
```
        # | Query | DB sequence | PK |
        # |-------|-------------|----|
        # |   1   |     1       |  2 | # First item returned in the first query to the DB
        # |   1   |     2       |  3 | # Second item returned in the first query to the DB
        # |   1   |     3       |  4 |
        # |   2   |     1       |  5 | # First item returned in the second query to the DB
        # |   2   |     2       |  6 | # Second item returned in the second query to the DB
        ...
```

Similarly, `Ret Sequence` is then denoting the position in the array of results actually returned from the backend to the client.

Reply:
Got it, helpful. Thank you!

And re: client side filtering. That makes sense... I was going a little wild trying to find where that was happening. :)